src/server/socket.test.ts

```diff
@@ -345,5 +345,102 @@ describe("Server-side socket testing", () => {
+      const { address, port } = server.address() as AddressInfo;
+      socket_client = Client(
+        `http://[${address}]:${port}/rooms/a418c099-4114-4c55-8a5b-4a142c2b26d1`,
+        {
+          autoConnect: false,
+        }
+      );
+
+      socket_admin = Client(
+        `http://[${address}]:${port}/rooms/a418c099-4114-4c55-8a5b-4a142c2b26d1/admin`,
+        {
+          autoConnect: false,
+        }
+      );
+
+      const initialized = Promise.all([
+        allEvents(socket_admin, ["QuestionNew", "connect", "RoomJoined"]),
+        allEvents(socket_client, ["QuestionNew", "connect", "RoomJoined"]),
+      ]);
+
+      socket_client.connect();
+      socket_admin.connect();
+
+      // Wait for both clients to be fully initialized
+      await initialized;
+
+      // Specify expected shape of the question object using Jest matchers
+      const question_matcher = {
+        id: expect.any(Number),
+        question: "A new question",
+        anonymous: true,
+        approved: false,
+        votes: 0,
+        created_at: expect.any(String),
+      };
+
+      // The admin channel should receive the question for approval
+      const admin_events = new Promise<Question[]>((resolve) => {
+        socket_admin.off("QuestionNew").on("QuestionNew", resolve);
+      }).then((questions) => {
+        expect(questions).toHaveLength(1);
+        expect(questions[0]).toMatchObject(question_matcher);
+        return questions[0].id;
+      });
+
+      // Send in a new question as a participant
+      const participant_callbacks = new Promise((resolve) => {
+        socket_client.emit(
+          "QuestionAsk",
+          { question: "A new question", anonymous: true },
+          resolve
+        );
+      }).then((success) => {
+        expect(success).toBe(true);
+      });
+
+      // Wait for the question to be asked and sent for approval
+      const [questionId] = await Promise.all([
+        admin_events,
+        participant_callbacks,
+      ]);
+
+      // If the admin approves the question it should be sent to the participant(s)
+      const participant_events = new Promise<Question[]>((resolve) => {
+        socket_client.off("QuestionNew").on("QuestionNew", resolve);
+      }).then((questions) => {
+        expect(questions).toHaveLength(1);
+        expect(questions[0]).toMatchObject({
+          ...question_matcher,
+          approved: true,
+        });
+      });
+
+      const admin_callbacks = new Promise<Question>((resolve) => {
+        socket_admin.emit("QuestionApprove", { questionId }, resolve);
+      }).then((question) => {
+        expect(question).toMatchObject({ ...question_matcher, approved: true });
+      });
+
+      // Wait for the question to be approved and sent to the participant(s)
+      await Promise.all([participant_events, admin_callbacks]);
+
+      // If the admin removes the question it should be removed from all views
```

Comment:
@KatieMacalintal One approach to reducing the duplication might be to integrate this into to the test above, that is the original test would include submitting, approving and deleting. While that is a very involved test, that also reduces the duplicated setup.